openfeature_flagsmith/provider.py

```diff
@@ -0,0 +1,134 @@
+import json
+import typing
+from json import JSONDecodeError
+from numbers import Number
+
+from flagsmith.exceptions import FlagsmithClientError
+from flagsmith.flagsmith import Flagsmith
+from openfeature.evaluation_context import EvaluationContext
+from openfeature.exception import ErrorCode
+from openfeature.flag_evaluation import FlagResolutionDetails, FlagType, Reason
+from openfeature.provider import Metadata
+from openfeature.provider.provider import AbstractProvider
+
+_BASIC_FLAG_TYPE_MAPPINGS = {
+    FlagType.BOOLEAN: bool,
+    FlagType.INTEGER: int,
+    FlagType.FLOAT: float,
+    FlagType.STRING: str,
+}
+
+
+class FlagsmithProvider(AbstractProvider):
+    def __init__(
+        self,
+        client: Flagsmith,
+        use_boolean_config_value: bool = False,
+        return_value_for_disabled_flags: bool = False,
+        use_flagsmith_defaults: bool = False,
+    ):
```

Comment:
Not sure if any of this benefits from being moved to the `initialize` method: https://github.com/open-feature/python-sdk/blob/7ba7d6146f0f801cadfd7593dc6df4b7d4f488d4/openfeature/provider/__init__.py#L11

Reply:
Yeah, this is interesting, I actually deliberated on this for a while. 

I guess they could be moved to the initialize method, BUT it's an anti pattern in python to define class attributes outside of the `__init__` method, so I'd end up defining the defaults in the `__init__` method and then overriding them in the `initialize` method, something like:

```python
        def __init__(
        self,
        client: Flagsmith,
    ):
        self._client = client
        self.return_value_for_disabled_flags = False
        ...

    def initialize(self, evaluation_context: EvaluationContext) -> None:
        self.return_value_for_disabled_flags = evaluation_context.attributes.get(
            "return_value_for_disabled_flags", self.return_value_for_disabled_flags
        )
        ...
```

I'm definitely keen to reduce the customisation that I've had to make to the __init__ method so I'd love some feedback on this one. 

So I guess the issue I'm trying to solve is kind of two fold: 
 1. I'm using the Flagsmith client here to not have to duplicate code between the OF provider and our SDK
 2. There are differences regarding the way in which our SDK works, compared to the OF SDK, which means we need some extra 'mapping' configuration.

It's these 'mapping' configurations that I'm trying to slot in somewhere and, in my mind at least, the `__init__` method feels like the right place for those. 

Sorry for the ramble. Thoughts @toddbaert @federicobond ?

Reply:
After thinking about it a bit, and doing some of my own research with regards to the Pythonic idiom recommending that class attributes are defined in __init__, I'm inclined to agree with your current approach. I think this makes the most sense.